pkg/types/metrics.go

```diff
@@ -141,3 +138,28 @@ type Statistics struct {
 func (s *Statistics) SetDefaults() {
 	s.RecentErrors = 10
 }
+
+// OtelGRPC provides configuration settings for the gRPC open-telemetry.
+type OtelGRPC struct {
+	Endpoint string            `description:"Sets the gRPC endpoint (host:port) of the collector." json:"endpoint,omitempty" toml:"endpoint,omitempty" yaml:"endpoint,omitempty"`
+	Insecure bool              `description:"Disables client transport security for the exporter." json:"insecure,omitempty" toml:"insecure,omitempty" yaml:"insecure,omitempty" export:"true"`
+	TLS      *ClientTLS        `description:"Defines client transport security parameters." json:"tls,omitempty" toml:"tls,omitempty" yaml:"tls,omitempty" export:"true"`
+	Headers  map[string]string `description:"Headers sent with payload." json:"headers,omitempty" toml:"headers,omitempty" yaml:"headers,omitempty" export:"true"`
+}
+
+// SetDefaults sets the default values.
+func (c *OtelGRPC) SetDefaults() {
+	c.Endpoint = "localhost:4317"
+}
+
+// OtelHTTP provides configuration settings for the HTTP open-telemetry.
+type OtelHTTP struct {
+	Endpoint string            `description:"Sets the HTTP endpoint (scheme://host:port/path) of the collector." json:"endpoint,omitempty" toml:"endpoint,omitempty" yaml:"endpoint,omitempty"`
+	TLS      *ClientTLS        `description:"Defines client transport security parameters." json:"tls,omitempty" toml:"tls,omitempty" yaml:"tls,omitempty" export:"true"`
+	Headers  map[string]string `description:"Headers sent with payload." json:"headers,omitempty" toml:"headers,omitempty" yaml:"headers,omitempty" export:"true"`
```

Comment:
Here I am wondering if we should export the headers as they could contain sensitive data?

Reply:
Good point, I think we shouldn't. I'll remove the export struct tag.

Reply:
@kevinpollet done.